### PR TITLE
Add type param for PUT entity in v2

### PR DESCRIPTION
**Pull request by crbrox**
@kzangeli 
@fgalan 

 Add: type param for PUT entity in v2

#988 #992 #1000

### CODE REVIEW

File: test/functionalTest/cases/0988_put_with_param_type/0988_put_with_param_type.test [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    "id": "E1",
+    "type": "T2",
+    "A1":{
+         "value": "V1 A1 E1/T2"
+    },
+    "A2":{
+         "value": "V1 A2 E1/T2"
+    }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
```
Comment (kzangeli) [inline test/functionalTest/cases/0988_put_with_param_type/0988_put_with_param_type.test]: What happens if this payload (where type is set to T2) is sent with URI param type=T5 ?

Hint: 400 Bad Request (discussion in NGSIv2 needed to decide the behavior) ...

With the current implementation I believe the URI param silently overwrites the value in the payload.

Comment (kzangeli) [inline test/functionalTest/cases/0988_put_with_param_type/0988_put_with_param_type.test]: Awaiting a decision in this matter we leave this *as is* for possible future modification.

NTC

**Comment by kzangeli** [comment]
LGTM

**Comment by crbrox** [comment]
CNR fixed at 687aed1

**Pull request merged**